CatAnalyzer/python/analyzers/ntuple_cfi.py

```diff
@@ -0,0 +1,64 @@
+import FWCore.ParameterSet.Config as cms
+
+ntuple = cms.EDAnalyzer("GenericNtupleMaker",
+    failureMode = cms.untracked.string("error"),
+    #failureMode = cms.untracked.string("skip"),
+    eventCounters = cms.vstring("nEventsTotal"),
+    int = cms.PSet(
+        vertex_n = cms.InputTag("catVertex", "nGoodPV"),
+        cutstep = cms.InputTag("eventSel", "cutstep"),
+    ),
+    float = cms.PSet(
+        met_pt = cms.InputTag("eventSel", "met"),
+        met_phi = cms.InputTag("eventSel", "metphi"),
+    ),
+    ints = cms.PSet(),
+    floats = cms.PSet(),
+    cands = cms.PSet(
```

Comment:
This "cands" reads any objects in the event compatible w/ reco::Candidate using the expression parser in CMSSW. This means you can do almost any variable associated with the candidate object - triggers, reference checking, basic functions, etc.. maybe lambdas?